crates/stc_ts_file_analyzer/src/analyzer/types/mod.rs

```diff
@@ -675,15 +675,20 @@ impl Analyzer<'_, '_> {
         let is_str = types.iter().any(|ty| ty.is_str());
         let is_num = types.iter().any(|ty| ty.is_num());
         let is_bool = types.iter().any(|ty| ty.is_bool());
-
+        let is_enum_variant = types.iter().any(|ty| ty.is_enum_variant());
+        
         if u32::from(is_str) + u32::from(is_num) + u32::from(is_bool) >= 2 {
             return never!();
         }
 
+        if u32::from(is_str) + u32::from(is_enum_variant) + u32::from(is_bool) >= 2 {
```

Comment:
I think this is wrong, because enum can be a number or string.

Reply:
I want you to look at this.
```ts
const enum Tag1 {}
const enum Tag2 {}
const enum Tag3 {
    f = "f",
    d = "d"
}

declare let s1: string & Tag1; // never
declare let s2: string & Tag2; // never
declare let s3: number & Tag1 & Tag2; // number & Tag1 & Tag2
declare let s4: boolean & Tag2; // never
declare let s5: Tag1 & Tag2; // Tag1 & Tag2
declare let s6: string & Tag3; // Tag3

s1 = s2;
s2 = s3; // error
s3 = s4;
s4 = s1;

declare let s7: true & false; // never
```

enum can intersect number, but can't intersect string;

Let's consider situations like s6.

Reply:
Even in that case, the current code is problematic because `number & NumericEnum` should not be never

Reply:
Yeah, I checked in the playground and what I said is correct.
`s6` is `Tag3` because `Tag3` Is a subset of `string`

Reply:
Will the following three tests be sufficient?

```ts
const enum Tag1 {
  a = "a",
  b = "b",
}
const enum Tag2 {
  A,
  B,
  C,
  D
}

declare let s1: string & Tag1;
declare let s2: string;

s1 = Tag1.a;
s2 = s1;


declare let s3 : number & Tag2;
declare let s4 : number;

s3 = Tag2.A;
s4 = s3;
```

Reply:
Yeap, but you should run `check.sh` anyway, and it will detect all the regressions